jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java

```diff
@@ -380,10 +384,165 @@ public void testInclude() throws Exception
+        //be set during an include
+        String responses = _connector.getResponse("""
+            GET /context/IncludeServlet?do=hello HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+            \r
+            """);
+
+        responses = responses.replaceFirst("Content-Length: .*\r\n", "");
+
+        String expected = """
+            HTTP/1.1 200 OK\r
+            Connection: close\r
+            \r
+            Include:
+            Hello
+            ---
+            """;
+
+        assertEquals(expected, responses);
+    }
+
+    @Test
+    public void testIncludeWriterOutputStream() throws Exception
+    {
+        _contextHandler.addServlet(IncludeServlet.class, "/IncludeServlet/*");
+        _contextHandler.addServlet(AssertIncludeServlet.class, "/AssertIncludeServlet/*");
+
+        //test include, along with special extension to include that allows headers to
+        //be set during an include
+        String responses = _connector.getResponse("""
+            GET /context/IncludeServlet?do=assertinclude&do=more&test=1&headers=true HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+            \r
+            """);
+
+        String expected = """
+            HTTP/1.1 200 OK\r
+            specialSetHeader: specialSetHeader\r
+            specialAddHeader: specialAddHeader\r
+            Content-Length: 20\r
+            Connection: close\r
+            \r
+            Include:
+            INCLUDE---
+            """;
+
+        assertEquals(expected, responses);
+    }
+
+    @Test
+    public void testIncludeStatic() throws Exception
+    {
+        _contextHandler.addServlet(IncludeServlet.class, "/IncludeServlet/*");
+        _contextHandler.addServlet(new ServletHolder("default", DefaultServlet.class), "/");
+        _server.start();
+
+        String responses = _connector.getResponse("""
+            GET /context/IncludeServlet?do=static HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+            \r
+            """);
+
+        String expected = """
+            HTTP/1.1 200 OK\r
+            Content-Length: 31\r
+            Connection: close\r
+            \r
+            Include:
+            Test 2 to too two
+            ---
+            """;
+
+        assertEquals(expected, responses);
+    }
+
+    @Test
+    public void testIncludeStaticWithWriter() throws Exception
+    {
+        _contextHandler.addServlet(new ServletHolder(new IncludeServlet(true)), "/IncludeServlet/*");
+        _contextHandler.addServlet(new ServletHolder("default", DefaultServlet.class), "/");
+        _server.start();
+
+        String responses = _connector.getResponse("""
+            GET /context/IncludeServlet?do=static HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+            \r
+            """);
+
+        String expected = """
+            HTTP/1.1 200 OK\r
```

Comment:
What's the reason why using the writer results in the absence of the `Content-Length` header?

Reply:
The setting of a content-length header is highly dependent on the HTTP version, the request version, the request headers and most importantly when/how the response is committed.    We don't want to test if HttpConnection and HttpGenerator are doing the right framing here, that is tested elsewhere.  We just want to see content and/or headers